### PR TITLE
Add support for Mobilenet v2 in Image Classification transfer learning

### PR DESCRIPTION
**Pull request by ashbhandare**
The base MobileNetV2 model is from https://github.com/tensorflow/models/tree/master/research/slim/nets/mobilenet . Checkpoint file used: https://storage.googleapis.com/mobilenet_v2/checkpoints/mobilenet_v2_1.4_224.tgz

Benchmarked transfer learning on the Cats V/s Dogs dataset at:
https://www.microsoft.com/en-us/download/details.aspx?id=54765

Trained last layer for 10 epochs with 0.0001f learning rate with batch size 32.
Final accuracy on eval set:
 Micro-accuracy: 0.980959936533122,macro-accuracy = 0.980874311779911
Comparable Tensorflow acuracy for same task: 
https://www.tensorflow.org/tutorials/images/transfer_learning#train_the_model

### CODE REVIEW

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 1]
```diff
@@ -1249,7 +1249,7 @@ public void TensorFlowImageClassification()
             var pipeline = mlContext.Transforms.LoadImages("Image", fullImagesetFolderPath, false, "ImagePath") // false indicates we want the image as a VBuffer<byte>
                 .Append(mlContext.Model.ImageClassification(
                     "Image", "Label",
-                    // Just by changing/selecting InceptionV3 here instead of 
+                    // Just by changing/selecting InceptionV3/MobilenetV2 here instead of
```
Comment (eerhardt) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: Do you want to add a test for the new functionality? The CI isn't executing any of your new code.

Comment (ashbhandare) [reply test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: Added unit test.

---
In reply to: [336686355](https://github.com/dotnet/machinelearning/pull/4351#discussion_r336686355) [](ancestors = 336686355)

File: src/Microsoft.ML.Dnn/DnnCatalog.cs [hunk 2]
```diff
@@ -144,7 +144,7 @@ public static class DnnCatalog
         {
             var options = new ImageClassificationEstimator.Options()
             {
-                ModelLocation = arch == Architecture.ResnetV2101 ? @"resnet_v2_101_299.meta" : @"InceptionV3.meta",
+                ModelLocation = ModelLocation[arch],
```
Comment (codemzs) [inline src/Microsoft.ML.Dnn/DnnCatalog.cs]: >ModelLocation = ModelLocation[arch], [](start = 16, length = 36)

Not your fault but since we are here might as well correct it. ModelLocation is not something that we need as an input from the user hence it should not be part of the options class. PLease remove this and move it as a property of ImageClassificationTransformer but not need to serialize it. This would mean you will need to move download code from lines 172 to 205 into ImageClassification constructor and specifically when we are not loading the model from disk, i,.e we are training the model for the first i.e within if (loadModel == false) case. This also means we can move ImagePreprocessingSize and ModelLocation constants into ImageClassificationTransformer since they are not really a property that needs to be set for initializing ImageClassification Estimator or something we need to expose publically.

Comment (ashbhandare) [reply src/Microsoft.ML.Dnn/DnnCatalog.cs]: Since this change is not realted to the current PR adding MobilenetV2 support, I will make this change in the PR fixing [https://github.com/dotnet/machinelearning/issues/4307](https://github.com/dotnet/machinelearning/issues/4307).

---
In reply to: [337218608](https://github.com/dotnet/machinelearning/pull/4351#discussion_r337218608) [](ancestors = 337218608)

Comment (codemzs) [reply src/Microsoft.ML.Dnn/DnnCatalog.cs]: ok, thanks @ashbhandare !
